numpyro/infer/autoguide.py

```diff
@@ -864,6 +865,373 @@ def _single_sample(_rng_key):
+    Annealed Importance Sampling (DAIS) [2, 3] to infer local latent variables.
+    Unlike :class:`AutoDAIS` this guide can be used in conjunction with data subsampling.
+    Note that the resulting ELBO can be understood as a particular realization of a
+    `locally enhanced bound' as described in reference [4].
+
+    **References:**
+
+    1. *Surrogate Likelihoods for Variational Annealed Importance Sampling*,
+       Martin Jankowiak, Du Phan
+    2. *MCMC Variational Inference via Uncorrected Hamiltonian Annealing*,
+       Tomas Geffner, Justin Domke
+    3. *Differentiable Annealed Importance Sampling and the Perils of Gradient Noise*,
+       Guodong Zhang, Kyle Hsu, Jianing Li, Chelsea Finn, Roger Grosse
+    4. *Variational Inference with Locally Enhanced Bounds for Hierarchical Models*,
+       Tomas Geffner, Justin Domke
+
+    Usage::
+
+        def global_model():
+            return numpyro.sample("theta", dist.Normal(0, 1))
+
+        def local_model(theta):
+            with numpyro.plate("data", 8, subsample_size=2):
+                tau = numpyro.sample("tau", dist.Gamma(5.0, 5.0))
+                numpyro.sample("obs", dist.Normal(0.0, tau), obs=jnp.ones(2))
+
+        model = lambda: local_model(global_model())
+        base_guide = AutoNormal(global_model)
+        guide = AutoSemiDAIS(model, local_model, base_guide, K=4)
+        svi = SVI(model, guide, ...)
+
+    :param callable model: A NumPyro model.
+    :param callable local_model: The portion of `model` that includes the local latent variables only.
+        The signature of `local_model` should be the return type of the global model with global latent
+        variables only.
+    :param callable base_guide: A guide for the global latent variables, e.g. an autoguide.
+        The return type should be a dictionary of latent sample sites and corresponding samples.
+    :param str prefix: A prefix that will be prefixed to all param internal sites.
+    :param int K: A positive integer that controls the number of HMC steps used.
+        Defaults to 4.
+    :param float eta_init: The initial value of the step size used in HMC. Defaults
+        to 0.01.
+    :param float eta_max: The maximum value of the learnable step size used in HMC.
+        Defaults to 0.1.
+    :param float gamma_init: The initial value of the learnable damping factor used
+        during partial momentum refreshments in HMC. Defaults to 0.9.
+    :param float init_scale: Initial scale for the standard deviation of the variational
+        distribution for each (unconstrained transformed) local latent variable. Defaults to 0.1.
+    """
+
+    def __init__(
+        self,
+        model,
+        local_model,
+        base_guide,
+        *,
+        prefix="auto",
+        K=4,
+        eta_init=0.01,
+        eta_max=0.1,
+        gamma_init=0.9,
+        init_scale=0.1,
+    ):
+        # TODO does the value of init_loc_fn used here matter at all?
```

Comment:
this is only used to setup `_local_potential_fn_gen` right? i.e. the value of `init_loc_fn` doesn't matter?

Reply:
that's right